ledger/participant-integration-api/src/main/scala/platform/indexer/MeteringAggregator.scala

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.MeteringStore
+import com.daml.ledger.participant.state.index.v2.MeteringStore.ParticipantMetering
+import com.daml.ledger.resources.ResourceOwner
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.indexer.MeteringAggregator.{toOffsetDateTime, toTimestamp}
+import com.daml.platform.store.appendonlydao.SqlExecutor
+import com.daml.platform.store.backend.MeteringParameterStorageBackend.LedgerMeteringEnd
+import com.daml.platform.store.backend.{
+  MeteringParameterStorageBackend,
+  MeteringStorageWriteBackend,
+  ParameterStorageBackend,
+}
+
+import java.sql.Connection
+import java.time.temporal.ChronoUnit
+import java.time.{OffsetDateTime, ZoneOffset}
+import java.util.{Timer, TimerTask}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object MeteringAggregator {
+
+  class Owner(
+      meteringStore: MeteringStorageWriteBackend,
+      parameterStore: ParameterStorageBackend,
+      meteringParameterStore: MeteringParameterStorageBackend,
+      metrics: Metrics,
+      period: FiniteDuration = 6.minutes,
+  ) {
+
+    private[platform] def apply(
+        sqlExecutor: SqlExecutor
+    )(implicit loggingContext: LoggingContext): ResourceOwner[Unit] = {
+      val aggregator = new MeteringAggregator(
+        meteringStore,
+        parameterStore,
+        meteringParameterStore,
+        metrics,
+        sqlExecutor,
+      )
+      for {
+        _ <- ResourceOwner.forFuture(() => aggregator.initialize())
+        _ <- ResourceOwner.forTimer(() => new Timer()).map { timer =>
+          timer.scheduleAtFixedRate(
+            new TimerTask {
+              override def run(): Unit = {
+                Await.ready(aggregator.run(), 60.minutes)
```

Comment:
while I see `aggregator.run()` failures are handled inside, but this `Await` may throw as well and it wont bee seen in any logging, or is there a general timer handler?

I suspect `60.minutes`  timeout also deserves to be parametrised or at least a named constant.

Reply:
Additionally: 60 minutes seems enough, but if for some reason not: it would mean to this algorithm that the next scheduled task will step on the long-running ones toes. Which we probably don't want to happen neither.

Reply:
Good call, await wrapped in error logging try.